metricbeat/helper/server/http/http_test.go

```diff
@@ -120,17 +166,16 @@ func checkServerReady(host string, port int) error {
 
 }
 
-func getHTTPServer(t *testing.T, host string, port int, connectionType string) (server.Server, error) {
+func getHTTPServer(t *testing.T, host string, port int, connectionType string, handleFunc func(writer http.ResponseWriter, req *http.Request)) (server.Server, error) {
```

Comment:
```suggestion
func getHTTPServer(t *testing.T, host string, port int, connectionType string, handleFunc http.HandleFunc) (server.Server, error) {
```

Reply:
will not be an issue with `NewHttpServerWithHandler`